src/compiler/checker.ts

```diff
@@ -33771,6 +33777,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 declaration.kind !== SyntaxKind.Constructor &&
                 declaration.kind !== SyntaxKind.ConstructSignature &&
                 declaration.kind !== SyntaxKind.ConstructorType &&
+                !(isJSDocSignature(declaration) && getJSDocRoot(declaration)?.parent?.kind === SyntaxKind.Constructor) &&
                 !isJSDocConstructSignature(declaration) &&

```

Comment:
Is `isJSDocConstructSignature` supposed to do the check?

Reply:
Confusingly, JSDocConstructSignature refers to closure type syntax: `function(new: SomeClass)` is equivalent to TS `new (): SomeClass`.

The naming scheme could be better though.